proof/crefine/AARCH64/VSpace_C.thy

```diff
@@ -3047,6 +3047,164 @@ lemma vcpu_switch_ccorres:
         (vcpuSwitch v) (Call vcpu_switch_'proc)"
   by (cases v; clarsimp simp: vcpu_switch_ccorres_None[simplified] vcpu_switch_ccorres_Some[simplified])
 
+lemma modify_armHSCurVCPU_split:
+  "modifyArchState (armHSCurVCPU_update (\<lambda>_. p)) = do modifyArchState (armHSCurVCPU_update f);
+                                                      modifyArchState (armHSCurVCPU_update (\<lambda>_. p))
+                                                   od"
+  apply (clarsimp simp: modifyArchState_def modify_modify)
+  apply (rule ext)
+  apply (rule_tac f="\<lambda>t. modify t s" for s in arg_cong)
+  apply (rule ext)
+  apply (case_tac "ksArchState s")
+  apply clarsimp
+  done
+
+lemma modify_armHSCurVCPU_when_split:
+  "modifyArchState (armHSCurVCPU_update (\<lambda>_. p)) = do when P (modifyArchState (armHSCurVCPU_update f));
+                                                      modifyArchState (armHSCurVCPU_update (\<lambda>_. p))
+                                                   od"
+  apply (cases P; clarsimp)
+  apply (subst modify_armHSCurVCPU_split[of _ f])
+  apply simp
+  done
+
+lemma modifyArchState_armHSCurVCPU_Skip:
+  "ccorres dc xfdc (\<lambda>s. armHSCurVCPU (ksArchState s) = curvcpu) UNIV hs
+     (modifyArchState (armHSCurVCPU_update (\<lambda>_. curvcpu))) SKIP"
+  apply (clarsimp simp: modifyArchState_def)
+  apply (rule ccorres_from_vcg)
+  apply (rule allI, rule conseqPre, vcg)
+  apply (clarsimp simp: simpler_gets_def simpler_modify_def)
+  apply (clarsimp simp: rf_sr_def cstate_relation_def Let_def)
+  apply (clarsimp simp: carch_state_relation_def carch_globals_def cur_vcpu_relation_def)
+  done
+
+(* FIX ARMHYP MOVE *)
+lemma armHSCurVCPU_update_active_false_ccorres:
```

Comment:
`FIX` -> `FIXME:` (otherwise grep won't find it)

(or, of course, you could move it now :))